src/lib/intersection.js

```diff
@@ -0,0 +1,21 @@
+const intersection = function() {
+  var newArray = [];
+  var workingArray = arguments[0];
+
+  for(let a = 1; a < arguments.length; a++){
+    newArray = [];
```

Comment:
Just by looking at this code, I can't tell why you reset `newArray` to an empty array here?  It's declared above, initialized as an empty array - is there a need to reset it every time you iterate through this outer loop?  If so, I wonder if there's a way to make it more explicit what is happening through variable naming.